CppImport/src/macro/StaticStuff.cpp

```diff
@@ -0,0 +1,358 @@
+/***********************************************************************************************************************
+ **
+ ** Copyright (c) 2011, 2015 ETH Zurich
+ ** All rights reserved.
+ **
+ ** Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+ ** following conditions are met:
+ **
+ **    * Redistributions of source code must retain the above copyright notice, this list of conditions and the
+ **      following disclaimer.
+ **    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+ **      following disclaimer in the documentation and/or other materials provided with the distribution.
+ **    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+ **      derived from this software without specific prior written permission.
+ **
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ ** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+
+#include "StaticStuff.h"
+
+namespace CppImport {
+
+void StaticStuff::orderNodes(QVector<Model::Node*>& input)
+{
+	qSort(input.begin(), input.end(),
+			[](Model::Node* e1, Model::Node* e2)
+	{
+		if (auto commonAncestor = e1->lowestCommonAncestor(e2))
+			if (auto list = DCast<Model::List>(commonAncestor))
+			{
+				int index1 = -1;
+				for (auto c : list->children())
+					if (c == e1 || c->isAncestorOf(e1))
+					{
+						index1 = list->indexOf(c);
+						break;
+					}
+
+				int index2 = -1;
+				for (auto c : list->children())
+					if (c == e2 || c->isAncestorOf(e2))
+					{
+						index2 = list->indexOf(c);
+						break;
+					}
+
+				return index1 < index2;
+			}
+
+		return true;
```

Comment:
This method is incorrect. The order should be transitive

suppose that
- A is not in the same list as B
- B in earlier in the same list as C

your function will always correctly compare (b,c) and (c,b), but I can get the following contradiction:
compare(A,B) = true; therefore A<B
compare(B,C) = true; therefore B<C
compare(C,A) = true; therefore C<A

You need to define an order based on the position in the tree, or otherwise restrict how this method is used.